### DISCUSSION TITLE
Selectors as Func<T, TResult>

### DISCUSSION THREAD

**Discussion by cmpbedes** [byme8/ZeroQL#78] [Q&A] [answered]
I'm writing an abstraction over the ZeroQL-generated code and was wondering if there's a way to specify the output of the query as a parameter instead, something like this:

``` csharp
//lambda approach:

Task<User> GetMyUser()
{
  return _client.Query(q => q.Me(x => new { x.FirstName, x.Surname }));
}

//Func approach

Task<T> GetMyUser<T>(Func<User, T> selector)
{
  return _client.Query(q => q.Me(selector));
}
```

**Comment by byme8** [comment, marked as answer]
No, it is impossible to use it like that because the source generator
wouldn't be able to figure out the selector in compile time.

**Reply by cmpbedes** [reply]
I had a feeling this would be the case, thanks for confirming
